pkg/server/handler.go

```diff
@@ -225,6 +229,65 @@ func WithInClusterServiceAccountRequestRewrite(handler http.Handler) http.Handle
 	})
 }
 
+// WithServiceAccountRewrite replaces the user info of the context for service
+// accounts by representing them as globally valid kcp service account representation
+// with a warrant in the old format (to match kube-like (cluster) role bindings).
+func WithServiceAccountRewrite(handler http.Handler) http.Handler {
```

Comment:
Can service accounts be split too to separate PR? Just enable service account for cross-workspaces?

Reply:
Will try. Have started the rebase now.